cassandra/client.go

```diff
@@ -322,21 +331,23 @@ func addSslOptions(cluster *gocql.ClusterConfig, options *sslOptions) *gocql.Clu
 	return cluster
 }
 
-func initializeSession(cluster *gocql.ClusterConfig) *gocql.Session {
+func initializeSession(cluster *gocql.ClusterConfig, co clientOptions) *gocql.Session {
 	session, err := cluster.CreateSession()
 	if err != nil {
 		log.Fatal(err.Error())
 	}
 
-	if err := session.Query(createKeyspaceCQL).Exec(); err != nil {
-		log.Fatal(err.Error())
+	if co.createKeyspace {
```

Comment:
No it's not :/ if you try to create keyspace and you don't have permissions to your cassandra cluster then your snap task will fail b/c cassandra publisher will panic b/c gocql will exit with panic ;). This flag is simply to handle situation when you know that you don't have permissions.

If you've got permissions to create keyspace, then you're right, but you cannot simply predict if remote cluster give you required permissions or not